openvdb_points/tools/IndexFilter.h

```diff
@@ -267,7 +261,11 @@ class RandomLeafFilter
         return mNextIndex == index;
     }
 
+protected:
+    friend class ::TestIndexFilter;
```

Comment:
It would be nice if this class didn't know anything specific about the unit test classes - the unit tests should be free to evolve without having to go back and change the core libs. It looks like we're relying on the friend class to provide access to mLeafMap - have you considered removing the friend relationship in favour of making mLeafMap protected, and relying on class inheritance in the unit tests to initialise this data?